Support for waiting sequentically

Right now I can use the `Wait.forAll`-Stategy to wait for multiple conditions to be ready.
This does however check for all conditions in parallel.
For my usecase I need a sequence of things to be checked.

My usecase is a postgres DB that is up for init, then shuts down for the actual DB to come up.
So my Wait strategy is:

1. Wait for the log message: `PostgreSQL init process complete; ready for start up.`
2. Wait for the log message:  `database system is ready to accept connections`
3. `Wait.forListeningPorts()`

I can implement this change, but there are multiple variants on how to do that.

1.  Alter `Wait.forAll()` to accept an optional parameter: `Wait.forAll([], "ExecuteInSequence")` or `Wait.forAll([], "ExecuteInParallel" )` where `"ExecuteInParallel"` is used by default, to keep compatibility
2. Create a new Wait method, like `Wait.forAllInSequence()`

What would be your preferred solution?

I just cloned the repo and ran the tests and there are a lot of test-failures on my machine (Apple M2 Pro, 32 GB Ram, running colima).
Is there more todo than `npm install` and then `npm test`?

```
$ colima version
colima version 0.8.1
git commit: 96598cc5b64e5e9e1e64891642b91edc8ac49d16

runtime: docker
arch: aarch64
client: v28.0.4
server: v27.4.0
```
My colima startup command: `colima start --arch aarch64 --vm-type=vz --vz-rosetta --cpu 6 --memory 12 --disk 150`

Part of the test output:
```
packages/testcontainers/src/wait-strategies/health-check-wait-strategy.test.ts > HealthCheckWaitStrategy > should wait for health check
TypeError: fetch failed
 ❯ checkContainerIsHealthy packages/testcontainers/src/utils/test-helper.ts:13:20
     11| export const checkContainerIsHealthy = async (container: StartedTestContainer): Promise<void> => {
     12|   const url = `http://${container.getHost()}:${container.getMappedPort(8080)}`;
     13|   const response = await fetch(`${url}/hello-world`);
       |                    ^
     14|   expect(response.status).toBe(200);
     15| };
 ❯ packages/testcontainers/src/wait-strategies/health-check-wait-strategy.test.ts:18:5

{
  stack: 'AggregateError: \n' +
    '    at internalConnectMultiple (node:net:1118:18)\n' +
    '    at afterConnectMultiple (node:net:1685:7)',
  errors: [
    {
      stack: 'Error: connect ECONNREFUSED 127.0.0.1:35490\n' +
        '    at createConnectionError (node:net:1648:14)\n' +
        '    at afterConnectMultiple (node:net:1678:16)',
      message: 'connect ECONNREFUSED 127.0.0.1:35490',
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 35490,
      constructor: 'Function<Error>',
      name: 'Error',
      toString: 'Function<toString>'
    },
    {
      stack: 'Error: connect ECONNREFUSED ::1:35490\n' +
        '    at createConnectionError (node:net:1648:14)\n' +
        '    at afterConnectMultiple (node:net:1678:16)',
      message: 'connect ECONNREFUSED ::1:35490',
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '::1',
      port: 35490,
      constructor: 'Function<Error>',
      name: 'Error',
      toString: 'Function<toString>'
    }
  ],
  code: 'ECONNREFUSED',
  stackStr: 'AggregateError: \n' +
    '    at internalConnectMultiple (node:net:1118:18)\n' +
    '    at afterConnectMultiple (node:net:1685:7)',
  nameStr: 'AggregateError',
  expected: 'undefined',
  actual: 'undefined',
  message: '',
  constructor: 'Function<AggregateError>',
  name: 'Caused by: AggregateError',
  toString: 'Function<toString>',
  stacks: []
}
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { errors: [ { stack: 'Error: connect ECONNREFUSED 127.0.0.1:35490\n    at createConnectionError (node:net:1648:14)\n    at afterConnectMultiple (node:net:1678:16)', message: 'connect ECONNREFUSED 127.0.0.1:35490', errno: -61, code: 'ECONNREFUSED', syscall: 'connect', address: '127.0.0.1', port: 35490, constructor: 'Function<Error>', name: 'Error', toString: 'Function<toString>' }, { stack: 'Error: connect ECONNREFUSED ::1:35490\n    at createConnectionError (node:net:1648:14)\n    at afterConnectMultiple (node:net:1678:16)', message: 'connect ECONNREFUSED ::1:35490', errno: -61, code: 'ECONNREFUSED', syscall: 'connect', address: '::1', port: 35490, constructor: 'Function<Error>', name: 'Error', toString: 'Function<toString>' } ], code: 'ECONNREFUSED' }
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[26/42]⎯

 FAIL  packages/testcontainers/src/wait-strategies/log-wait-strategy.test.ts > LogWaitStrategy > should wait for log
TypeError: fetch failed
 ❯ checkContainerIsHealthy packages/testcontainers/src/utils/test-helper.ts:13:20
     11| export const checkContainerIsHealthy = async (container: StartedTestContainer): Promise<void> => {
     12|   const url = `http://${container.getHost()}:${container.getMappedPort(8080)}`;
     13|   const response = await fetch(`${url}/hello-world`);
       |                    ^
     14|   expect(response.status).toBe(200);
     15| };
 ❯ packages/testcontainers/src/wait-strategies/log-wait-strategy.test.ts:13:5

{
  stack: 'AggregateError: \n' +
    '    at internalConnectMultiple (node:net:1118:18)\n' +
    '    at afterConnectMultiple (node:net:1685:7)',
  errors: [
    {
      stack: 'Error: connect ECONNREFUSED 127.0.0.1:35677\n' +
        '    at createConnectionError (node:net:1648:14)\n' +
        '    at afterConnectMultiple (node:net:1678:16)',
      message: 'connect ECONNREFUSED 127.0.0.1:35677',
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 35677,
      constructor: 'Function<Error>',
      name: 'Error',
      toString: 'Function<toString>'
    },
    {
      stack: 'Error: connect ECONNREFUSED ::1:35677\n' +
        '    at createConnectionError (node:net:1648:14)\n' +
        '    at afterConnectMultiple (node:net:1678:16)',
      message: 'connect ECONNREFUSED ::1:35677',
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '::1',
      port: 35677,
      constructor: 'Function<Error>',
      name: 'Error',
      toString: 'Function<toString>'
    }
  ],
  code: 'ECONNREFUSED',
  stackStr: 'AggregateError: \n' +
    '    at internalConnectMultiple (node:net:1118:18)\n' +
    '    at afterConnectMultiple (node:net:1685:7)',
  nameStr: 'AggregateError',
  expected: 'undefined',
  actual: 'undefined',
  message: '',
  constructor: 'Function<AggregateError>',
  name: 'Caused by: AggregateError',
  toString: 'Function<toString>',
  stacks: []
}
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Serialized Error: { errors: [ { stack: 'Error: connect ECONNREFUSED 127.0.0.1:35677\n    at createConnectionError (node:net:1648:14)\n    at afterConnectMultiple (node:net:1678:16)', message: 'connect ECONNREFUSED 127.0.0.1:35677', errno: -61, code: 'ECONNREFUSED', syscall: 'connect', address: '127.0.0.1', port: 35677, constructor: 'Function<Error>', name: 'Error', toString: 'Function<toString>' }, { stack: 'Error: connect ECONNREFUSED ::1:35677\n    at createConnectionError (node:net:1648:14)\n    at afterConnectMultiple (node:net:1678:16)', message: 'connect ECONNREFUSED ::1:35677', errno: -61, code: 'ECONNREFUSED', syscall: 'connect', address: '::1', port: 35677, constructor: 'Function<Error>', name: 'Error', toString: 'Function<toString>' } ], code: 'ECONNREFUSED' }
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[27/42]⎯


 Test Files  17 failed | 75 passed (92)
      Tests  42 failed | 479 passed | 4 skipped (525)
   Start at  06:48:22
   Duration  516.41s (transform 2.94s, setup 0ms, collect 104.51s, tests 4969.71s, environment 15ms, prepare 8.38s)
```

--- a/packages/modules/cassandra/src/cassandra-container.ts
+++ b/packages/modules/cassandra/src/cassandra-container.ts
@@ -10,7 +10,9 @@ export class CassandraContainer extends GenericContainer {
 
   constructor(image = "cassandra:5.0.2") {
     super(image);
-    this.withExposedPorts(CASSANDRA_PORT).withWaitStrategy(Wait.forHealthCheck()).withStartupTimeout(120_000);
+    this.withExposedPorts(CASSANDRA_PORT)
+      .withWaitStrategy(Wait.forAll([Wait.forHealthCheck(), Wait.forListeningPorts()]))
+      .withStartupTimeout(120_000);
   }
 
   public withDatacenter(dc: string): this {

--- a/packages/modules/cockroachdb/src/cockroachdb-container.ts
+++ b/packages/modules/cockroachdb/src/cockroachdb-container.ts
@@ -11,7 +11,7 @@ export class CockroachDbContainer extends GenericContainer {
     super(image);
     this.withExposedPorts(COCKROACH_PORT, COCKROACH_HTTP_PORT)
       .withCommand(["start-single-node", "--insecure", `--http-addr=0.0.0.0:${COCKROACH_HTTP_PORT}`])
-      .withWaitStrategy(Wait.forHealthCheck());
+      .withWaitStrategy(Wait.forAll([Wait.forHealthCheck(), Wait.forListeningPorts()]));
   }
 
   public withDatabase(database: string): this {

--- a/packages/modules/postgresql/src/postgresql-container.ts
+++ b/packages/modules/postgresql/src/postgresql-container.ts
@@ -10,7 +10,7 @@ export class PostgreSqlContainer extends GenericContainer {
   constructor(image = "postgres:13.3-alpine") {
     super(image);
     this.withExposedPorts(POSTGRES_PORT);
-    this.withWaitStrategy(Wait.forHealthCheck());
+    this.withWaitStrategy(Wait.forAll([Wait.forHealthCheck(), Wait.forListeningPorts()]));
     this.withStartupTimeout(120_000);
   }
 

